### DISCUSSION TITLE
Overlay and configuration file

### DISCUSSION THREAD

**Discussion by srid** [srid/emanote#37] [Q&A]
Moved comment from @hi-learn-triz-and-usit at https://github.com/srid/emanote/issues/35#issuecomment-849259087 to here

```
# Run emanote live server on current directory, using common themes and public notes (3 layers)
cd ~/MyPrivateNotes
emanote run ~/myemanotethemes ~/MyPublicNotes ./
```
When _wouldn't_ we want to include public notes and common themes when running `emanote` in `MyPrivateNotes`? Because if the answer is "never", and in fact if it doesn't even make sense to do so (because some links or themes will be broken) shouldn't the first two layers be part of the configuration file of the third layer?

Basically, I'm suggesting that the overlay graph should be expressed using an implicit linked list (or more generally a dag) in the configuration files of the layers (each layer has the option to point to layers it depends on). This handles the common use case, where a site is built up of pieces, without complicating the interface. This does not handle the case where we want to easily be able to swap the pieces; my question is whether this is actually a thing users will want to do.

_Originally posted by @hi-learn-triz-and-usit in https://github.com/srid/emanote/issues/35#issuecomment-849259087_

**Comment by hi-learn-triz-and-usit** [comment]
I wouldn't frame it as "overlay vs conffile"; in both cases, the overlay system is essential. The difference is in how you specify the overlay structure.

**Reply by srid** [reply]
Write a concrete proposal then. (Because I must be missing something)

**Reply by hi-learn-triz-and-usit** [reply]
You asked for a proposal: https://github.com/srid/emanote/pull/41

I know it's a bit extra. Partially, I went overboard to show that I was serious; partially, it was fun.
